phoenix-core/src/main/java/org/apache/phoenix/coprocessor/MetaDataEndpointImpl.java

```diff
@@ -1222,7 +1228,7 @@ public void createTable(RpcController controller, CreateTableRequest request,
                 // exists without making an additional query
                 PTable table =
                         loadTable(env, tableKey, cacheKey, clientTimeStamp, HConstants.LATEST_TIMESTAMP);
-                if (table != null) {
+                if (table != null && !isTableDeleted(table)) {
```

Comment:
I reverted this change and ran all the tests and they passed, so I will revert this change.